src/ns_tracker/nsdeps.clj

```diff
@@ -21,8 +22,30 @@
 	     (contains? #{:use :require} (first form)))
     (apply union (map #(deps-from-libspec nil %) (rest form)))))
 
+(defn- find-resource [path dirs]
+  (->> (map #(File. ^File % ^String path) dirs)
+       (filter #(.isFile ^File %))
+       first))
+
+(defn- deps-from-ns-metadata [decl dirs]
+  (->> (or (meta (second decl))
```

Comment:
Let's be a little more precise here. So perhaps:

```clojure
(defn- ns-attr-map [[_ _ docstring? attr-map?]]
  (cond
    (map? docstring?) docstring?
    (map? attr-map?) attr-map?))

(defn- ns-metadata [[_ name :as decl]]
  (merge (meta name) (ns-attr-map decl))

(defn- deps-from-ns-metadata [decl dirs]
  (let [deps      (:ns-tracker/resource-deps (ns-metadata decl))
        resources (for [d deps] [d (find-resource d dirs)])]
    (print-warnings-for-missing-resources resources)
    (set (map second resources))))
```

Reply:
Good suggestion! That’ll make it much more readable.